### PR TITLE
Add CLI guides for listing and creating clusters

### PR DESCRIPTION
**Pull request by axbarsan**
Towards https://github.com/giantswarm/giantswarm/issues/18343

Nothing too crazy, following the spec. The cluster creation guide gets data such as the provider, organization and cluster name from the state (so when you refresh the page, you see that you get the same UID in the guide as you do in the form)

## Preview

<details>
<summary>List clusters</summary>

image

</details>

<details>
<summary>Create cluster</summary>

image

</details>

### CODE REVIEW

File: src/components/MAPI/clusters/guides/CreateClusterGuide.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+              href: docs.kubectlGSInstallationURL,
+              external: true,
+            },
+            {
+              label: 'kubectl gs template cluster command',
+              href: docs.kubectlGSTemplateClusterURL,
+              external: true,
+            },
+            {
+              label: 'Cluster CRD schema',
```
Comment (axbarsan) [inline src/components/MAPI/clusters/guides/CreateClusterGuide.tsx:43]: Also added a link to the `Cluster` CRD, to keep it consistent with the organizations guides

**Comment by marians** [comment]
Checking now

**Comment by marians** [comment]
As discussed in SIG UX, let's add `--context CONTEXT` here already. I think the flag should appear as far to the left as possible. E. g.

    kubectl gs --context gs-gremlin get clusters --all-namespaces

**Comment by marians** [comment]
In the cluster creation form context, the idea was that the `kubectl gs template cluster` command in the guide adapts to the user's input in the form. I can't see that happening.

**Comment by axbarsan** [comment]
> In the cluster creation form context, the idea was that the `kubectl gs template cluster` command in the guide adapts to the user's input in the form. I can't see that happening.

The organization and name do adapt. Did you also want to include all the other flags that `kubectl gs` supports?

**Comment by marians** [comment]
> Did you also want to include all the other flags that kubectl gs supports?

Yes, all the flags that are relevant to the form displayed. (I'm not sure if that covers all flags or leaves some unused.)

File: src/components/MAPI/guides/utils.ts [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+export type KubectlGSCommandModifier = (
+  parts: readonly string[]
+) => readonly string[];
+
+/**
+ * Construct a `kubectl gs` command example.
+ *
+ * You can use modifiers to add arguments or flags to the command.
+ * */
+export function makeKubectlGSCommand(
```
Comment (axbarsan) [inline src/components/MAPI/guides/utils.ts:18]: I created a new builder function for creating `kubectl gs` examples.

It uses modifiers to add arguments/flags to the commands, and each command has a custom modifier. There's also a modifier for automatically formatting the command output, so it spans multiple lines.

**Comment by axbarsan** [comment]
I updated the code and now we have a (over-engineered) way to create `kubectl gs` command templates for guides. Please take a look ✌️

File: src/components/MAPI/guides/utils.ts [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+      hasArgs = true;
+      newParts.push(part);
+      // If the next value is not a flag, add an argument delimiter.
+      if (peekPart && !isFlag(peekPart)) {
+        newParts.push(' ');
+      }
+    }
+
+    if (hasFlags && !hasArgs) {
+      newParts.unshift(formattingLineBreak);
```
Comment (axbarsan) [inline src/components/MAPI/guides/utils.ts:207]: I hate this, but I guess it works for now

**Comment by marians** [comment]
All right, the good news first: It seems to be working. Nice!

Here for some improvements regarding cluster creation.

1. The idea was to keep the `template cluster` command as short as possible. So if a default is used, the according flag should not be applied. Specifically there is
  - `--control-plane-az 2` despite the selection being the default `Automatic selection` (which should be done in the backend as far as I understand). Here I'm actually not sure whether I should see this as a logical error or simply as a cosmetic detail. I don't know if this is even implemented properly in `kubectl gs template cluster`.
  - `--release 14.1.4` despite it being the default. This is mostly cosmetic, however there is a difference between using the command without a `--release` (which means: apply the default at the time of execution) and having a release set.

2. The `--context` flag.
  - Is not supported in the `template cluster` command, as I just learned. Probably as there is no API communication.
  - Should be present in the `kubectl apply` command in step 3.

3. Indentation of the commands is missing

None of these is blocking.

**Review by marians — APPROVED** [review]

**Comment by axbarsan** [comment]
> 1. The `--context` flag.
> 
> * Is not supported in the `template cluster` command, as I just learned. Probably as there is no API communication.
> * Should be present in the `kubectl apply` command in step 3.

Totally missed the `template cluster` one 😆 Updated the other one.

> 2. Indentation of the commands is missing

Fixed ✅

**Comment by axbarsan** [comment]
> * `--release 14.1.4` despite it being the default. This is mostly cosmetic, however there is a difference between using the command without a `--release` (which means: apply the default at the time of execution) and having a release set.

But the user sees the latest version selected in the cluster creation form (e.g. `15.0.0`). I would be confused if when I selected the latest release version, the command example would remove the `--release` flag. The user doesn't know that we use defaulting to set the latest version when `--release` is not used.

**Pull request merged**